### PR TITLE
clu #17565 Invoke OnMemberRemoved callback when  cluster.shutdown

### PR DESCRIPTION
**Pull request by patriknw**
- must also be done when the listener actor stops before the
  MemberRemoved event has been received
- add test for this
- clarify docs with example that shuts down actor system and
  exit jvm

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #17565

**Comment by patriknw** [comment]
PLS BUILD

File: akka-samples/akka-sample-cluster-java/src/main/java/sample/cluster/factorial/FactorialFrontendMain.java [hunk 1]
```diff
@@ -30,10 +34,22 @@ public void run() {
...
+      public void run() {
+        // exit JVM when ActorSystem has been terminated
+        final Runnable exit = new Runnable() {
+          @Override
+          public void run() {
+            System.exit(-1);
+          }
+        };
+        system.registerOnTermination(exit);
+        // if the ActorSystem does not terminate
```
Comment (ktoso) [inline akka-samples/akka-sample-cluster-java/src/main/java/sample/cluster/factorial/FactorialFrontendMain.java]: As mentioned elsewhere, I'd reword to  "in case ActorSystem shutdown takes longer than 10 seconds, exit the JVM forcefully anyway"?

**Comment by ktoso** [comment]
LGTM otherwise :)

**Comment by drewhk** [comment]
LGTM

File: akka-samples/akka-sample-cluster-java/src/main/java/sample/cluster/factorial/FactorialFrontendMain.java (hunk 1 above)
Comment (patriknw) [inline akka-samples/akka-sample-cluster-java/src/main/java/sample/cluster/factorial/FactorialFrontendMain.java]: +1
